library/kani/src/lib.rs

```diff
@@ -154,8 +148,22 @@ fn any_raw_inner<T>() -> T {
 #[inline(never)]
 #[rustc_diagnostic_item = "KaniPanic"]
 #[doc(hidden)]
-pub const fn panic(message: &'static str) -> ! {
-    panic!("{}", message)
+pub const fn panic(_message: &'static str) -> ! {
+    __kani_dummy_never()
+}
+
+/// This is a dummy function that is intended to be called from functions with
+/// `rustc_diagnostic_item`, i.e. that have kani hooks, to make sure they
+/// compile.
+/// While we could use `unreachable!()` or `panic!()` as the body of this
+/// function, both cause Kani to produce a warning on missing `caller_location`
+/// intrinsic (see https://github.com/model-checking/kani/issues/2010) and
+/// produce unnecessarily large MIR.  This function is handled via a hook
+/// anyway, so we just need to put a body that rustc does not complain about. An
+/// infinite loop works out nicely.
+const fn __kani_dummy_never() -> ! {
```

Comment:
`#[doc(hidden)]` for things we don't want customer-facing.

Reply:
Is this needed for a private function?

Reply:
Oh, good point. Nevermind then :)